cmd/util/ledger/migrations/staged_contracts_migration.go

```diff
@@ -148,11 +154,179 @@ func (m *StagedContractsMigration) InitMigration(
 	return nil
 }
 
+// collectAndRegisterStagedContractsFromPayloads scan through the payloads and collects the contracts
+// staged through the `MigrationContractStaging` contract.
+func (m *StagedContractsMigration) collectAndRegisterStagedContractsFromPayloads(allPayloads []*ledger.Payload) error {
+
+	// If the contracts are already passed as an input to the migration
+	// then no need to scan the storage.
+	if len(m.contractsByLocation) > 0 {
+		return nil
+	}
+
+	var stagingAccount string
+
+	switch m.chainID {
+	case flow.Testnet:
+		stagingAccount = "0x2ceae959ed1a7e7a"
+	case flow.Mainnet:
+		stagingAccount = "0x56100d46aa9b0212"
+	default:
+		// For other networks such as emulator etc. no need to scan for staged contracts.
+		m.log.Warn().Msgf("staged contracts are not collected for %s state", m.chainID)
+		return nil
+	}
+
+	stagingAccountAddress := common.Address(flow.HexToAddress(stagingAccount))
+
+	// Filter-in only the payloads belong to the staging account.
+	stagingAccountPayloads := make([]*ledger.Payload, 0)
+	for _, payload := range allPayloads {
+		key, err := payload.Key()
+		if err != nil {
+			return err
+		}
+
+		address := flow.BytesToAddress(key.KeyParts[0].Value)
+
+		if stagingAccountAddress == common.Address(address) {
```

Comment:
Nit: Avoid yoda condition:

```suggestion
		if common.Address(address) == stagingAccountAddress {
```